File: sorting my life (sort algarithoms)/insertion-sort.js

let Insertion_Sort = (arr) => {
    for (let i = 0; i < arr.length; i++){
        let key = arr[i]
        let j = i -1
        while( j >= 0 && arr[j] > key){
            arr[j + 1] = arr[j]
            j = j - 1
        }
        arr[j + 1] = key
        
    }
    return arr
}

let testdata = [5, 2, 4, 6, 1, 3]
console.log(Insertion_Sort(testdata))

File: sorting my life (sort algarithoms)/insertion-sort.dart

var data = [1,4,8,3,1,4,123,2];

List insertionsort(arr){
  for (var index = 0; index < arr.length; index++) {
    var key = arr[index];
    var backindex = index - 1;
    while (backindex >= 0 && key < arr[backindex]) {
      arr[backindex + 1] = arr[backindex];
      backindex = backindex -1;
    }
   }
return arr;
}

void main(){
  print(insertionsort(data));
}